core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -322,8 +332,9 @@ public Node start() {
 
         validateNodeBeforeAcceptingRequests(settings, transportService.boundAddress());
 
-        DiscoveryNode localNode = injector.getInstance(DiscoveryNodeService.class)
-                .buildLocalNode(transportService.boundAddress().publishAddress());
+        String nodeId = injector.getInstance(NodeEnvironment.class).nodeId();
```

Comment:
Can we just assign `NodeEnvironment` next to `Environment` and don't do the guice lookup?